core/src/main/java/com/linecorp/armeria/server/VirtualHostBuilder.java

```diff
@@ -478,7 +501,8 @@ public VirtualHostBuilder serviceUnder(String pathPrefix, HttpService service) {
      * @throws IllegalArgumentException if the specified path pattern is invalid
      */
     public VirtualHostBuilder service(String pathPattern, HttpService service) {
-        service(Route.builder().path(pathPattern).build(), service);
+        final String prefixed = contextPath == null ? pathPattern : contextPath + pathPattern;
+        service(Route.builder().path(prefixed).build(), service);
```

Comment:
Instead of adding the prefix here, what do you think of adding the prefix at `VirtualHostBuilder#build`?

If we add routes here there are several problems:
1) Some builders like `VirtualHostBuilder#route()`, or `VirtualHostBuilder#annotatedService`, `VirtualHostBuilder#serviceUnder` won't be applied
2) If the `contextPath` value changes, then the applied prefix can be inconsistent

Reply:
thanks. understood!